feat: update `Livechat_enabled_when_agent_idle` setting to not include `offline` agents in the query (#39495)
RocketChat/Rocket.Chat: .changeset/purple-boxes-shout.md

---
'@rocket.chat/models': patch
'@rocket.chat/meteor': patch
---

> Fixes an issue where offline livechat agents were being assigned to visitors.

Reviewer: Yep. The problem is that this behavior has been like this for quite a while. Probably since years ago. We treated it as "the right behavior". The fact we're changing the understanding now (or fixing, idk) doesn't change the fact this is a behavior change.

> Fixes an issue where offline livechat agents were being assigned to visitors.

Author: The changeset change was addressed on https://github.com/RocketChat/Rocket.Chat/pull/39495/changes/b19afad8313778537aef52ea42804c8c4d3033fe

The comment for general understanding of the query was added on https://github.com/RocketChat/Rocket.Chat/pull/39495/changes/c75d663d496e280a32981f7e22192f9c7bc2e555

> Updates the behavior of the `Livechat_enabled_when_agent_idle` setting. When enabled, the routing query now excludes agents with an offline status, ensuring visitors are not assigned to them.

Reviewer: ```suggestion
Updates the behavior of the `Livechat_enabled_when_agent_idle` setting. When enabled, the routing query now excludes `offline` agents, ensuring no new conversations are assigned to them.
```
